source/Irrlicht/CMakeLists.txt

```diff
@@ -393,6 +411,43 @@ add_library(IrrlichtMt
 	$<TARGET_OBJECTS:IRRGUIOBJ>
 )
 
+# Alias target provides add_submodule compatibility
+add_library(IrrlichtMt::IrrlichtMt ALIAS IrrlichtMt)
+
+target_include_directories(IrrlichtMt
+	PUBLIC
+		"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/>"
+		"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+		"$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/irrlichtmt>"
+	PRIVATE
+		${PNG_INCLUDE_DIR}
+		"$<$<BOOL:OGL_ENABLED>:${OPENGL_INCLUDE_DIR}>"
+		"$<$<BOOL:OGLES2_ENABLED>:${OPENGLES2_INCLUDE_DIR}>"
+		"$<$<BOOL:OGLES2_ENABLED>:${EGL_INCLUDE_DIR}>"
+		"$<$<BOOL:ANDROID>:${ANDROID_NDK}/sources/android/native_app_glue}>"
+		"$<$<BOOL:REQUIRE_X11>:${X11_INCLUDE_DIR}>"
+)
+
+target_link_libraries(IrrlichtMt
+	PRIVATE
+		ZLIB::ZLIB
+		JPEG::JPEG
+		"${PNG_LIBRARY}"
+
+		"$<$<BOOL:OGL_ENABLED>:${OPENGL_LIBRARIES}>"
+		"$<$<BOOL:OGLES1_ENABLED>:${OPENGLES_LIBRARY}>"
+		"$<$<BOOL:OGLES1_ENABLED>:${EGL_LIBRARY}>"
```

Comment:
EGL is needed for both gles1 and gles2

I'm wondering would there be any issue with just putting `"${EGL_LIBRARY}"` here? CMake will ignore it if not set

Reply:
Good point, I'll fix that.